File: program/program/find-the-trace-of-a-matrix/find_the_trace_of_a_matrix.dart
import 'dart:io';

void main() {
  int n;
  print("Enter the number of rows and columns of matrix:");
  n = int.parse(stdin.readLineSync()!);
  List<List<int>> matrix = List.generate(n, (_) => List.filled(n, 0));
  print("Enter the elements of matrix:");
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
      matrix[i][j] = int.parse(stdin.readLineSync()!);
    }
  }
  int trace = 0;
  for (int i = 0; i < n; i++) {
    trace += matrix[i][i];
  }
  print("Trace of matrix is $trace");
}
 /* Output :
Enter the number of rows and columns of matrix:
3
Enter the elements of matrix:
1
2
3
4
5
6
7
8
9
Trace of matrix is 15. 
*/

File: program/program/find-the-trace-of-a-matrix/FindTheTraceOfAMatrix.ts
function trace(matrix: number[][]): number {
  let trace = 0;
  for (let i = 0; i < matrix.length; i++) {
    trace += matrix[i][i];
  }
  return trace;
}

const matrix = [
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9],
];
console.log(trace(matrix)); // Output: 15
